app/views/map/map.html.erb

```diff
@@ -21,53 +20,63 @@
     height: 80vh;
     background:rgba(255,255,255,0.5);
   }
+  .alert {
+    z-index: 600;
+  }
   </style>
 
-  <div id="map"></div>
+<div id="map"></div>
 
    <script>
 
-      var urlHash = urlHash();
+      var urlHash = urlMapHash();
+
+        // if there are location params in URL let them override, otherwise save lat/lon/zoom from controller
+        if (!urlHash.getUrlHashParameter('lat') || !urlHash.getUrlHashParameter('lon')) {
+          <% if @lat && @lon %>
+            urlHash.setUrlHashParameter('lat', <%= @lat %> + "");
+            urlHash.setUrlHashParameter('lon', <%= @lon %> + "");
+            <% if @zoom %>
+              urlHash.setUrlHashParameter('zoom', <%= @zoom %> + "");
+            <% end %>
+          <% end %>
+        }
+
       const params = urlHash.getUrlHashParameters();
      
+      var lat = parseFloat(params.lat);
+      var lon = parseFloat(params.lon);
+      var zoom = parseFloat(params.zoom);
+
       const bounds = new L.LatLngBounds(
         new L.LatLng(84.67351257, -172.96875),
         new L.LatLng(-54.36775852, 178.59375)
       );
 
-      var user_lat = null;
```

Comment:
Yes awesome i wanted this to be removed from here and not the URLHash logic! Thanks!